canopen/profiles/p402.py

```diff
@@ -423,13 +422,12 @@ def op_mode(self, mode):
             while self.op_mode != mode:
                 if time.monotonic() > timeout:
                     raise RuntimeError(
-                        "Timeout setting node {0}'s new mode of operation to {1}.".format(
-                            self.id, mode))
-            logger.info('Set node {n} operation mode to {m}.'.format(n=self.id, m=mode))
+                        f"Timeout setting node {self.id}'s new mode of operation to {mode}.")
+            logger.info('Set node %s operation mode to %s.', self.id, mode)
         except SdoCommunicationError as e:
-            logger.warning('[SDO communication error] Cause: {0}'.format(str(e)))
+            logger.warning('[SDO communication error] Cause: %s', e)
         except (RuntimeError, ValueError) as e:
-            logger.warning('{0}'.format(str(e)))
+            logger.warning(str(e))
```

Comment:
I'm not expert in DS402, so I'm not completely sure what may trigger these exceptions. I can see the `RuntimeError` a few lines up, but I don't know what can cause a `RuntimeError`. While at it: Both of these exceptions swallow the error. Is that intentional? It removes the user from being able to deal with it (except for log warnings).

Reply:
Not quite sure about the `ValueError`, but it could be catching any `NoneType` access there.  The `RuntimeError` shouldn't be caught, as the docstring explicitly lists it as a possible exception.  But that's material for another PR I'd say.

Reply:
Yes, please. I'm resolving this conversation.